examples/matrix_multiplier/tests/test_matrix_multiplier.py

```diff
@@ -0,0 +1,143 @@
+from cocotb.regression import TestFactory
+from cocotb.scoreboard import Scoreboard
+from cocotb.triggers import RisingEdge, ReadOnly
+
+NUM_SAMPLES = int(os.environ.get('NUM_SAMPLES', 3000))
+DATA_WIDTH = int(os.environ['DATA_WIDTH'])
+A_ROWS = int(os.environ['A_ROWS'])
+B_COLUMNS = int(os.environ['B_COLUMNS'])
+A_COLUMNS_B_ROWS = int(os.environ['A_COLUMNS_B_ROWS'])
+
+
+class MatrixMonitor(BusMonitor):
+    """Base class for monitoring inputs/outputs of Matrix Multiplier."""
+    def __init__(self, dut, callback=None, event=None):
+        super().__init__(dut, "", dut.clk_i, callback=callback, event=event)
+
+
+class MatrixInMonitor(MatrixMonitor):
+    """Monitor inputs to Matrix Multiplier module.
+
+    Generate expected results for each multiplication operation.
+    """
+    _signals = {"A": "a_i", "B": "b_i", "valid": "valid_i"}
+
+    async def _monitor_recv(self):
+        while True:
+            await RisingEdge(self.clock)
+            await ReadOnly()
+
+            if self.bus.valid.value:
+                a_matrix = self.bus.A.value
+                b_matrix = self.bus.B.value
+
+                # Calculate the expected result of C
+                c_expected = [
+                    BinaryValue(
+                        sum(
+                            [
+                                a_matrix[(i*A_COLUMNS_B_ROWS)+n] * b_matrix[(n*B_COLUMNS)+j]
+                                for n in range(A_COLUMNS_B_ROWS)
+                            ]
+                        ),
+                        n_bits=(DATA_WIDTH * 2) + math.ceil(math.log2(A_COLUMNS_B_ROWS)),
+                        bigEndian=False
+                    )
+                    for i in range(A_ROWS)
+                    for j in range(B_COLUMNS)
+                ]
+
+                self._recv(c_expected)
+
+
+class MatrixOutMonitor(MatrixMonitor):
+    """Monitor outputs from Matrix Multiplier module.
+
+    Capture result matrix for each multiplication operation.
+    """
+    _signals = {"C": "c_o", "valid": "valid_o"}
+
+    async def _monitor_recv(self):
+        while True:
+            await RisingEdge(self.clock)
+            await ReadOnly()
+
+            if self.bus.valid.value:
+                c_actual = self.bus.C.value
+                self._recv(c_actual)
+
+
+async def test_multiply(dut, a_data, b_data):
+    """Test multiplication of many matrices."""
+
+    cocotb.fork(Clock(dut.clk_i, 5, units='ns').start())
+
+    # Configure Scoreboard to compare module results to expected
+    expected_output = []
+
+    def add_expected(transaction):
+        expected_output.append(transaction)
+
+    in_monitor = MatrixInMonitor(dut, callback=add_expected)
+    out_monitor = MatrixOutMonitor(dut)
+
+    scoreboard = Scoreboard(dut)
+    scoreboard.add_interface(out_monitor, expected_output)
+
+    # Initial values
+    dut.valid_i <= 0
+    dut.a_i <= next(gen_a(1, lambda x: 0))
+    dut.b_i <= next(gen_b(1, lambda x: 0))
```

Comment:
This is not particularly readable IMO. See the following comments for the full suggestion.
```suggestion
    dut.a_i <= gen_a(lambda x: 0)
    dut.b_i <= gen_b(lambda x: 0)
```

Reply:
Added `create_a` and `create_b` to improve readability.